cmake/spheral_cxx-config.cmake.in

```diff
@@ -0,0 +1,51 @@
+
+@PACKAGE_INIT@
+
+set(ENABLE_CXXONLY "@ENABLE_CXXONLY@")
+set(SPHERAL_ENABLE_CUDA "@ENABLE_CUDA@")
+set(SPHERAL_ENABLE_MPI "@ENABLE_MPI@")
+set(SPHERAL_ENABLE_OPENMP "@ENABLE_OPENMP@")
+# TODO: Add check to avoid setting these when unnecessary
+set(CMAKE_C_COMPILER "@CMAKE_C_COMPILER@" CACHE PATH "")
+set(CMAKE_CXX_COMPILER "@CMAKE_CXX_COMPILER@" CACHE PATH "")
+set(CMAKE_Fortran_COMPILER "@CMAKE_Fortran_COMPILER@" CACHE PATH "")
+set(MPI_C_COMPILER "@MPI_C_COMPILER@" CACHE PATH "")
+set(MPI_CXX_COMPILER "@MPI_CXX_COMPILER@" CACHE PATH "")
+set(MPI_Fortran_COMPILER "@MPI_Fortran_COMPILER@" CACHE PATH "")
+set(MPIEXEC_EXECUTABLE "@MPIEXEC_EXECUTABLE@" CACHE PATH "")
+set(MPIEXEC_NUMPROC_FLAG "@MPIEXEC_NUMPROC_FLAG@" CACHE STRING "")
+set(-DMPI_C_COMPILER "@-DMPI_C_COMPILER@" CACHE PATH "")
+set(-DMPI_CXX_COMPILER "@-DMPI_CXX_COMPILER@" CACHE PATH "")
```

Comment:
I don't entirely understand why we are forwarding all of this compiler / MPI information, why is that?

Reply:
Since almost everything is handled as target exports, this file can probably be reduced significantly. I will look into it.